File: android/app/src/main/java/jagtheora/vorbis/VorbisBlock.java
package jagtheora.vorbis;

import jagtheora.misc.SimplePeer;
import jagtheora.ogg.OggPacket;

public final class VorbisBlock extends SimplePeer {

    public VorbisBlock(DSPState arg0) {
        this.init(arg0);
        if (this.b()) {
            throw new IllegalStateException();
        }
    }

    public int synthesis(OggPacket arg0) { return 0; }

    public void init(DSPState arg0) {}

    protected void clear() {}
}

File: src/jvmMain/kotlin/jagtheora/vorbis/VorbisBlock.kt
package jagtheora.vorbis

import jagtheora.misc.SimplePeer
import jagtheora.ogg.OggPacket

actual class VorbisBlock actual constructor(arg0: DSPState?) : SimplePeer() {
    init {
        this.init(arg0)
        check(!this.b())
    }

    actual external override fun clear()

    /*private*/ actual external fun init(arg0: DSPState?)

    actual external fun synthesis(arg0: OggPacket?): Int
}
